spec/stream_spec.rb

```diff
@@ -598,9 +598,25 @@
       expect(s1.remote_window).to eq 900
     end
 
-    it 'should keep track of incoming flow control' do
+    it 'should not update window when data received is under the threshold' do
       data = DATA.deep_dup
       datalen = data[:payload].bytesize
+      expect(@stream).not_to receive(:send) do |frame|
+        expect(frame[:type]).to eq :window_update
+        expect(frame[:increment]).to eq datalen
+      end
+      expect(@client).not_to receive(:send) do |frame|
+        expect(frame[:type]).to eq :window_update
+        expect(frame[:increment]).to eq datalen
+      end
+      @stream.receive HEADERS.deep_dup
+      @stream.receive data
+    end
+
+    it 'should update window when data received is over the threshold' do
+      data1 = DATA.merge(payload: 'a'*16_384, flags: [])
```

Comment:
Ditto, it's not clear from the test what the threshold is / how this logic is triggered.